core/src/main/java/com/epam/esm/dao/impl/GiftCertificateDaoImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.util.QueryBuildHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.epam.esm.constant.Query.*;
+
+@Repository
+public class GiftCertificateDaoImpl implements GiftCertificateDao {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final RowMapper<GiftCertificate> rowMapper;
+
+    @Autowired
+    public GiftCertificateDaoImpl(JdbcTemplate jdbcTemplate, RowMapper<GiftCertificate> rowMapper) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.rowMapper = rowMapper;
+    }
+
+    @Override
+    public void create(GiftCertificate giftCertificate) {
+        jdbcTemplate.update(CREATE_GIFT_CERTIFICATE, giftCertificate.getName()
+                ,giftCertificate.getDescription(), giftCertificate.getPrice()
+                ,giftCertificate.getDuration());
+    }
+
+    @Override
+    public List<GiftCertificate> getAll() {
+        return jdbcTemplate.query(GET_ALL_GIFT_CERTIFICATES, rowMapper);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        jdbcTemplate.update(DELETE_GIFT_CERTIFICATE_BY_ID, id);
+    }
+
+    @Override
+    public Optional<GiftCertificate> getById(Long id) {
+        return jdbcTemplate.query(GET_GIFT_CERTIFICATE_BY_ID, rowMapper, id).stream().findAny();
```

Comment:
seems that query should return only 1 row or null. I do not think that streams are useful here